### PR TITLE
P1 38 virtual wallet scheme

### PR DESCRIPTION
**Pull request by aagorelik**

### CODE REVIEW

File: lib/vchain_client.rb [hunk 1]
```diff
@@ -375,7 +375,7 @@ def add_data_point(point_type, input, weight = 1)
 
       data_point_factory = VChainClient::DataPoints::Factory.new(@config)
 
-      blockchain_facade = VChainClient::Blockchain::Facade.new(@config)
+      virtual_wallet = VChainClient::VirtualWallet::VirtualWallet.new(@config)
```
Comment (baitcode) [inline lib/vchain_client.rb]: ugly

File: lib/vchain_client.rb [hunk 2]
```diff
@@ -395,7 +395,7 @@ def add_data_point(point_type, input, weight = 1)
 
       document_to_send = document.export_for_data_point()
 
-      blockchain_tx_ids = blockchain_facade.write( ipfs_locator )
+      blockchain_tx_ids = virtual_wallet.write( ipfs_locator )
```
Comment (baitcode) [inline lib/vchain_client.rb]: are we ever going to distinguish between different blockchains? I'm not saying we should forsee everything here, ipfs_locator looks pretty concrete now, we can add some sot of prefixes in future to distinguish. Just pointing out for you to think

File: lib/vchain_client/virtual_wallet/adapters/bitcoind/bitcoind.rb [hunk 3]
```diff
@@ -83,15 +79,20 @@ def read(txid)
...
+
+        end
       end
 
       if raw_tx != nil
 
         if raw_tx.key?("confirmations")
 
-          if raw_tx["confirmations"] > confirmations_threshold
+          if raw_tx["confirmations"] >= confirmations_threshold
```
Comment (baitcode) [inline lib/vchain_client/virtual_wallet/adapters/bitcoind/bitcoind.rb]: How do we handle situation when threshold is low?

Comment (aagorelik) [inline lib/vchain_client/virtual_wallet/adapters/bitcoind/bitcoind.rb]: it is a matter of client' logic here - they sometimes can choose to lower the threshold.

**Pull request merged**